R/tab-book.R

```diff
@@ -158,18 +168,36 @@ setMethod("initialize", "MultitableResult", function (.Object, ...) {
                 name="Total"
             )
         )))
+    # .Object$result <- lapply(.Object$result, CrunchCube)
     .Object$result <- lapply(.Object$result, function (cube) {
         cube <- CrunchCube(cube)
-        ## If cubes are 3D (categorical array x multitable), aperm the cubes so
-        ## that column is multitable var (3 -> 2), row is category of
-        ## array (2 -> 1), subvar is "tab" (1 -> 3)
+        ## TODO: refactor with CrunchCubep-native methods (eg, dimensions<-, aperm)
         if (length(dim(cube)) == 3L) {
-            ## TODO: CrunchCube should get a dimensions<-
-            cube@dims <- CubeDims(cube@dims[c(2, 3, 1)])
-            cube@arrays <- lapply(cube@arrays, aperm, perm=c(2, 3, 1))
+            ## check if there is an MR, in which case there are actually 4 dims 
+            ## underlyingly, not 3 dims
+            selecteds <- is.selectedDimension(cube@dims)
+            if (any(which(selecteds) %in% c(3, 4))) {
+                ## the selected dimension is in the second half of the cube, so 
+                ## the MR is in the multitable
+                cube@dims <- CubeDims(cube@dims[c(2, 3, 4, 1)])
+                cube@arrays <- lapply(cube@arrays, aperm, perm=c(2, 3, 4, 1))
+            } else if (any(which(selecteds) %in% c(1, 2))) {
+                ## the selected dimension is in the first half of the cube, so 
+                ## the array is in the multitable
```

Comment:
Is it possible for a categorical array to be in the multitable? I don't know how I'd interpret that.

Reply:
Yup 🙃!